### DISCUSSION TITLE
Issue deploying to Vercel (MongoDB Database)

### DISCUSSION THREAD

**Discussion by LupuC** [prisma/prisma#23812] [Q&A] [answered]
### Question

Well, It looks like it doesn't connect to the database, when deploying on vercel.
Locally, on my PC, it works fine.
    504Gateway Timeout is the error.

### How to reproduce (optional)

1. Created a new NextJS project, added prisma, ran `prisma generate`, `prisma db push`. Database url set in .env file
2. Uploaded to github
3. Added to package, this postinstall script `"postinstall": "prisma generate"`
4. Created a project on Vercel.
5. Errors

### Expected behavior (optional)

I've set to Vercel Env. Table, a DATABASE_URL, to the mongodb url, and allowed it to PROD, PREVIEW and DEVELOPMENT.
Also, I've added the .env to github repo.

### Information about Prisma Schema, Client Queries and Environment (optional)

```prisma
generator client {
  provider = "prisma-client-js"
}

datasource db {
  provider = "mongodb"
  url      = env("DATABASE_URL")
}

//models here...
```

```ts
export async function GET(request: NextRequest) {
    try {
        const sections = await prisma.section.findMany({
            orderBy: { position: 'asc' },
            include: { tasks: {
                orderBy: { position: 'asc' },
            } },
        });

        return NextResponse.json({ sections });
    } catch (error) {
        return NextResponse.json({ error: 'Internal server error' }, { status: 500 });
    }
}

```

- OS: Vercel
- Database: MongoDB
- Node.js version: 20.x

- Run `prisma -v` to see your Prisma version and paste it
prisma                  : 5.12.1
@prisma/client          : 5.12.1
Computed binaryTarget   : windows
Operating System        : win32
Architecture            : x64
Node.js                 : v21.6.1
Query Engine (Node-API) : libquery-engine 473ed3124229e22d881cb7addf559799debae1ab (at node_modules\.pnpm\@prisma+engines@5.12.1\node_modules\@prisma\engines\query_engine-windows.dll.node)
Schema Engine           : schema-engine-cli 473ed3124229e22d881cb7addf559799debae1ab (at node_modules\.pnpm\@prisma+engines@5.12.1\node_modules\@prisma\engines\schema-engine-windows.exe)
Schema Wasm             : @prisma/prisma-schema-wasm 5.12.0-21.473ed3124229e22d881cb7addf559799debae1ab
Default Engines Hash    : 473ed3124229e22d881cb7addf559799debae1ab
Studio                  : 0.499.0

**Comment by ludralph** [comment, marked as answer]
Hi @LupuC 👋 

Thank you for raising this question.

Did you whitelist the requesting IP on the MongoDB side? The timeout is similar to this previous [discussion](https://github.com/prisma/prisma/discussions/21374).

**Reply by LupuC** [reply]
Yeah, that was it. Sorry for another post related to this, but I didn't find the discussion you provided. Thank you!
